### ISSUE TITLE
Idle timeout was exceeded

### ISSUE THREAD

**Issue by MatthK** [gutmensch/docker-dmarc-report#65]
I just tried to install this. The MySQL container is coming up and is healthy.
However, the dmarc-report container comes up as starting, and later then exits with code 1. The log shows the following:

```
CRIT: Dmarc reports could not be parsed. Check your IMAP and MYSQL Settings.
DEBUG: Parsing failed with the following output:
Started at Mon Dec  9 15:08:06 2024
Using Mail::IMAPClient version 3.43 on perl 5.034002
Connecting with IO::Socket::IP PeerAddr mail.domain.com PeerPort 993 Proto tcp Timeout 600 Debug 1
Connected to mail.domain.com
Read: 	* BYE Session terminated by server because idle timeout was exceeded
ERROR: * BYE Session terminated by server because idle timeout was exceeded at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 1439.
	Mail::IMAPClient::_get_response(Mail::IMAPClient=HASH(0xffffa468ff30), "*", "PREAUTH") called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 408
	Mail::IMAPClient::Socket(Mail::IMAPClient=HASH(0xffffa468ff30), IO::Socket::IP=GLOB(0xffffa468a478)) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 369
	Mail::IMAPClient::connect(Mail::IMAPClient=HASH(0xffffa468ff30)) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 315
	Mail::IMAPClient::new("Mail::IMAPClient", "Server", "mail.domain.com", "Port", 993, "Ssl", 0, "Starttls", ...) called at /usr/bin/dmarcts-report-parser.pl line 305
dmarcts-report-parser.pl: IMAP Failure: * BYE Session terminated by server because idle timeout was exceeded at /usr/bin/dmarcts-report-parser.pl line 305.
use tls without verify servercert.
connection to mail.domain.com with Ssl => 0, User => adr@domain.com, Ignoresizeerrors => 1
```

The mail box has emails in the Inbox folder, and "processed" and "error" folders exist as well.

Why can't it get the emails?

**Comment by gutmensch** [comment]
@MatthK Your IMAP connection details seem to be wrong, please mind
* Port 993 is usually IMAP over SSL (PARSER_IMAP_SSL=1)
* Port 143 can be plaintext or with STARTTLS (PARSER_IMAP_TLS=1)

At the moment due to the related perl connection definition madness you need to make sure to provide the correct connection details (seems you have PARSER_IMAP_SSL=0). https://github.com/gutmensch/docker-dmarc-report/tree/master?tab=readme-ov-file#optional-extended-configuration

**Comment by MatthK** [comment]
I'll try to change the port. 

However I tried to set the PARSER_IMAP_SSL=1. I then get an error stating that a secure connection is already in use.

**Comment by gutmensch** [comment]
Maybe try to verify connectivity to your imap server with openssl command and imap/starttls resp. tls settings (google for examples). It should tell you if e.g. verification failed or the server does something unpredicted on a port. The perl job and package are not making it super easy to debug.

**Comment by MatthK** [comment]
Changing the port to 143 did the trick. It now works like a charm.

The only thing is, when I processed the emails manually by running `docker compose exec dmarc-report /usr/bin/dmarcts-report-parser.pl -i -d -r=1` it did process them, however at the end it said: `dmarcts-report-parser.pl: Processed 0 emails.`

But never mind that, as the reports did show up and the emails were moved from the Inbox to the processed folder.
